app/main.py

```diff
@@ -1,3 +1,125 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import math
+from datetime import datetime
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def get_the_cost_of_the_trip(
+        customer: Customer,
+        shop: Shop,
+        fuel_price: int
+) -> int | float:
+    distance = math.dist(customer.location, shop.location) * 2
+    cost_of_a_trip = round(
+        (customer.car.fuel_consumption * fuel_price / 100) * distance, 2
+    )
+    return cost_of_a_trip
+
+
+def get_the_cost_of_the_products(
+        customer: Customer,
+        shop: Shop
+) -> int | float:
+    cost_of_the_products = 0
+    for name_of_product, number_of_products in customer.product_cart.items():
+        cost_of_the_products += (
+            number_of_products * shop.products[name_of_product]
+        )
+    return cost_of_the_products
+
+
+def get_check(data: dict, customer: Customer, the_cheapest_store: str) -> None:
+    the_cheapest_store_products = {}
+    today = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+    total_cost = 0
+
+    for shop_name in data.get("shops"):
+        if shop_name["name"] == the_cheapest_store:
+            the_cheapest_store_products.update(shop_name.get("products"))
+
+    print(
+        f"{today}\n"
+        f"Thanks, {customer.name}, for you purchase!\n"
+        f"You have bought:"
+    )
+
+    for name, price in the_cheapest_store_products.items():
+        total_cost += customer.product_cart[name] * price
+        print(
+            f"{customer.product_cart[name]} {name}s "
+            f"for {customer.product_cart[name] * price} dollars"
+        )
+    print(
+        f"Total cost is {total_cost} dollars\n"
+        f"See you again!\n"
+    )
+
+
+def shop_trip() -> None:
+    with open("config.json", "r") as file:
```

Comment:
```suggestion
    with open("app/config.json", "r") as file:
```
And test will be works

Reply:
Not fixed